common/djangoapps/student/models.py

```diff
@@ -332,6 +337,57 @@ def user_profile_pre_save_callback(sender, **kwargs):
         user_profile.profile_image_uploaded_at = None
 
 
+@receiver(post_save, sender=UserProfile)
+def user_profile_post_save_callback(sender, **kwargs):
+    """
+    Emit analytics events after saving the
+    """
+    user_profile = kwargs['instance']
+    # pylint: disable=protected-access
+    emit_field_changed_events(user_profile, user_profile.user, sender._meta.db_table)
+
+
+def emit_field_changed_events(instance, user, db_table):
```

Comment:
Must: This method should take an (optional) array of excluded_fields. For instance, UserProfile could have a meta field that we don't want to event, but User could also have a meta field, and maybe we want to event that one. This method shouldn't know anything about specific model fields.